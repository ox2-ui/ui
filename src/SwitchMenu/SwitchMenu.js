import PropTypes from 'prop-types';
import React from 'react';
import Button from '@ox2/button/Button';

/**
 * SwitchMenu Component
 */
const SwitchMenu = ({ onDismiss }) =>
  <div className="padding-b:40">
    <Button onClick={() => onDismiss()} btn="raised action">
      Done
    </Button>
    <div>
      SWITCH Lorem ipsum dolor sit amet, consectetur
      adipisicing elit. Itaque, sequi? Consequuntur
      inventore nulla ratione vitae obcaecati voluptatem
      animi molestiae ipsam et adipisci illo reiciendis
      saepe doloremque, quae perspiciatis iure vero.
    </div>
  </div>;

SwitchMenu.propTypes = {
  onDismiss: PropTypes.func.isRequired,
};

export default SwitchMenu;
