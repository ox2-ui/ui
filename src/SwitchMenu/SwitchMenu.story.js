import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SwitchMenu from './SwitchMenu';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('SwitchMenu', module)
  .add('default', () => (
    <SwitchMenu onDismiss={handler} />
  ));
