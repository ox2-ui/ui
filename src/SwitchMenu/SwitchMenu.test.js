/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SwitchMenu from './SwitchMenu';

const testHandler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <SwitchMenu onDismiss={testHandler} />,
  );

  expect(wrapper).toMatchSnapshot();
});
