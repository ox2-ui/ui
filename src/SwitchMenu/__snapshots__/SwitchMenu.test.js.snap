exports[`test renders correctly 1`] = `
<SwitchMenu
  onDismiss={[Function]}>
  <div
    className="padding-b:40">
    <Button
      btn="raised action"
      onClick={[Function]}>
      <button
        data-btn="raised action"
        onClick={[Function]}>
        Done
      </button>
    </Button>
    <div>
      SWITCH Lorem ipsum dolor sit amet, consectetur adipisicing elit. Itaque, sequi? Consequuntur inventore nulla ratione vitae obcaecati voluptatem animi molestiae ipsam et adipisci illo reiciendis saepe doloremque, quae perspiciatis iure vero.
    </div>
  </div>
</SwitchMenu>
`;
